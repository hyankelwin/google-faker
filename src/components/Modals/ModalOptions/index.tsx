import React from 'react';
import { IoApps, IoClose } from 'react-icons/io5';
import Modal from 'react-modal';
import {
  Image,
  ContentOptions,
  Content,
  ContentIconClose,
  Column,
  GoogleImage,
} from './styles';
import UserImage from '../../../assets/images/user.png';
import LogoGoogle from '../../../assets/images/logo-google.png';

interface ModalOptionsProps {
  isOpen: boolean;
  onCloseModal: () => void;
}

const ModalOptions: React.FC<ModalOptionsProps> = ({
  isOpen,
  onCloseModal,
}) => {
  return (
    <Modal isOpen={isOpen} ariaHideApp={false}>
      <Content>
        <ContentIconClose>
          <GoogleImage src={LogoGoogle} alt="Image of google search" />
          <IoClose
            data-testid="icon-close-modal"
            onClick={() => onCloseModal()}
            size={20}
          />
        </ContentIconClose>

        <ContentOptions>
          <Column>
            <p>Aplications</p>
            <IoApps size={20} />
          </Column>
          <Column>
            <p>Profile</p>
            <Image src={UserImage} alt="Image to simulate user logged" />
          </Column>
        </ContentOptions>
      </Content>
    </Modal>
  );
};

export default ModalOptions;
