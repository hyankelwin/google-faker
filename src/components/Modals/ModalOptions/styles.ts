import styled from 'styled-components';
import Color from '../../../styles/color';

export const Content = styled.div``;

export const ContentOptions = styled.div``;

export const ContentIconClose = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;

  svg {
    cursor: pointer;
  }
`;

export const Column = styled.div`
  width: 100%;
  background: ${Color.grey[200]};
  margin-bottom: 5px;
  height: 50px;

  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
`;

export const Image = styled.img`
  width: 40px;
  height: 40px;
  border-radius: 100%;
  border: 1px solid;
  border-color: ${Color.grey[200]};
`;

export const GoogleImage = styled.img`
  width: 80px;
`;
