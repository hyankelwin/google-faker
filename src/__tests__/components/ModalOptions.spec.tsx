import { fireEvent, render } from '@testing-library/react';
import { mount } from 'enzyme';
import ModalOptions from '../../components/Modals/ModalOptions';

describe('ModalOptions component', () => {
  describe('should be able to passed in all tests on index.tsx', () => {
    it('should be able to render ModalOptions component with success', () => {
      const props = {
        isOpen: true,
        onCloseModal: jest.fn(),
      };

      const wrapper = mount(<ModalOptions {...props} />);

      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should be able to simulate click on IconClose on ModalSearch component', () => {
      const props = {
        isOpen: true,
        onCloseModal: jest.fn(),
      };

      const { getAllByTestId } = render(<ModalOptions {...props} />);

      const iconCloseModal = getAllByTestId('icon-close-modal')[0];

      fireEvent.click(iconCloseModal);

      expect(iconCloseModal).toBeTruthy();
    });
  });
});
