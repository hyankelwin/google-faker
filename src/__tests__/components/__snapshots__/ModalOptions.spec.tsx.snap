// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalOptions component should be able to passed in all tests on index.tsx should be able to render ModalOptions component with success 1`] = `
"<ModalOptions isOpen={true} onCloseModal={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }}>
  <Modal isOpen={true} ariaHideApp={false} portalClassName=\\"ReactModalPortal\\" bodyOpenClassName=\\"ReactModal__Body--open\\" role=\\"dialog\\" closeTimeoutMS={0} shouldFocusAfterRender={true} shouldCloseOnEsc={true} shouldCloseOnOverlayClick={true} shouldReturnFocusAfterClose={true} preventScroll={false} parentSelector={[Function: parentSelector]} overlayElement={[Function: overlayElement]} contentElement={[Function: contentElement]}>
    <Portal containerInfo={{...}}>
      <ModalPortal defaultStyles={{...}} isOpen={true} ariaHideApp={false} portalClassName=\\"ReactModalPortal\\" bodyOpenClassName=\\"ReactModal__Body--open\\" role=\\"dialog\\" closeTimeoutMS={0} shouldFocusAfterRender={true} shouldCloseOnEsc={true} shouldCloseOnOverlayClick={true} shouldReturnFocusAfterClose={true} preventScroll={false} parentSelector={[Function: parentSelector]} overlayElement={[Function: overlayElement]} contentElement={[Function: contentElement]} style={{...}}>
        <div className=\\"ReactModal__Overlay\\" style={{...}} onClick={[Function (anonymous)]} onMouseDown={[Function (anonymous)]}>
          <div id={[undefined]} style={{...}} className=\\"ReactModal__Content\\" tabIndex=\\"-1\\" onKeyDown={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onClick={[Function (anonymous)]} role=\\"dialog\\" aria-label={[undefined]} aria-modal={true} data-testid={[undefined]}>
            <styled.div>
              <div className=\\"sc-aXZVg\\">
                <styled.div>
                  <div className=\\"sc-eqUAAy kJOtCY\\">
                    <styled.img src=\\"logo-google.png\\" alt=\\"Image of google search\\">
                      <img src=\\"logo-google.png\\" alt=\\"Image of google search\\" className=\\"sc-iGgWBj eAmVxJ\\" />
                    </styled.img>
                    <IoClose data-testid=\\"icon-close-modal\\" onClick={[Function: onClick]} size={20}>
                      <IconBase attr={{...}} data-testid=\\"icon-close-modal\\" onClick={[Function: onClick]} size={20}>
                        <svg stroke=\\"currentColor\\" fill=\\"currentColor\\" strokeWidth=\\"0\\" viewBox=\\"0 0 512 512\\" data-testid=\\"icon-close-modal\\" onClick={[Function: onClick]} className={[undefined]} style={{...}} height={20} width={20} xmlns=\\"http://www.w3.org/2000/svg\\">
                          <path d=\\"M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z\\" />
                        </svg>
                      </IconBase>
                    </IoClose>
                  </div>
                </styled.div>
                <styled.div>
                  <div className=\\"sc-gEvEer\\">
                    <styled.div>
                      <div className=\\"sc-fqkvVR jhkVIf\\">
                        <p>
                          Aplications
                        </p>
                        <IoApps size={20}>
                          <IconBase attr={{...}} size={20}>
                            <svg stroke=\\"currentColor\\" fill=\\"currentColor\\" strokeWidth=\\"0\\" viewBox=\\"0 0 512 512\\" className={[undefined]} style={{...}} height={20} width={20} xmlns=\\"http://www.w3.org/2000/svg\\">
                              <path d=\\"M104 160a56 56 0 1156-56 56.06 56.06 0 01-56 56zm152 0a56 56 0 1156-56 56.06 56.06 0 01-56 56zm152 0a56 56 0 1156-56 56.06 56.06 0 01-56 56zM104 312a56 56 0 1156-56 56.06 56.06 0 01-56 56zm152 0a56 56 0 1156-56 56.06 56.06 0 01-56 56zm152 0a56 56 0 1156-56 56.06 56.06 0 01-56 56zM104 464a56 56 0 1156-56 56.06 56.06 0 01-56 56zm152 0a56 56 0 1156-56 56.06 56.06 0 01-56 56zm152 0a56 56 0 1156-56 56.06 56.06 0 01-56 56z\\" />
                            </svg>
                          </IconBase>
                        </IoApps>
                      </div>
                    </styled.div>
                    <styled.div>
                      <div className=\\"sc-fqkvVR jhkVIf\\">
                        <p>
                          Profile
                        </p>
                        <styled.img src=\\"user.png\\" alt=\\"Image to simulate user logged\\">
                          <img src=\\"user.png\\" alt=\\"Image to simulate user logged\\" className=\\"sc-dcJsrY WaEVU\\" />
                        </styled.img>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </div>
        </div>
      </ModalPortal>
    </Portal>
  </Modal>
</ModalOptions>"
`;
